src/algebra/roption_instances.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2019 Hoang Le Truong. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author : Hoang Le Truong.
+
+roption α  can see as  Prop × α. (Prop, ∧) has natural algebra structure. Therefore if α is a semigroup, a comm_semigroup, 
+a comm_monoid,a add_semigroup, a add_comm_semigroup, a add_comm_monoid, a mul_action, a distrib_mul_action, so is roption α. 
+-/
+import   algebra.module data.pfun
+
+universes u v w
+variables {α : Type u} {β : Type v} {γ : Type w} 
+
+namespace roption
+noncomputable theory
+
+instance [has_zero α] : has_zero (roption α) := ⟨some (0 : α)⟩
+lemma zero_def [has_zero α] : (0 : roption α) = ⟨true, λ _, (0 : α)⟩ := rfl
+
+instance [has_one α] :  has_one (roption α)  := ⟨some (1 : α)⟩
+lemma one_def [has_one α] : (1 : roption α) = ⟨true, λ _, (1 : α)⟩ := rfl
+
+attribute [to_additive roption.has_zero] roption.has_one
+attribute [to_additive roption.zero_def] roption.one_def
+
+instance [has_add α] : has_add (roption α) := ⟨λ x y, ⟨x.dom ∧ y.dom, λ h, x.get (h.1)+ y.get (h.2)⟩⟩
+lemma add_def [has_add α] (x y : roption α) : x+y = ⟨x.dom ∧ y.dom, λ h, x.get (h.1)+ y.get (h.2)⟩ := rfl
+
+instance [has_mul α] : has_mul (roption α) := ⟨λ x y, ⟨x.dom ∧  y.dom, λ h, x.get (h.1) * y.get (h.2)⟩⟩
+lemma mul_def  [has_mul α] (x y : roption α) : x * y = ⟨x.dom ∧ y.dom , λ h,  x.get (h.1) * y.get (h.2)⟩ := rfl
+
+attribute [to_additive roption.has_add] roption.has_mul
+attribute [to_additive roption.add_def] roption.mul_def
+
+instance [has_neg α] : has_neg (roption α) := ⟨λ x, ⟨x.dom , λ h, -x.get h⟩⟩
```

Comment:
These should go. They don't obey any axioms.

Reply:
I will remove it